https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/19994/testReport/junit/(root)/Suite-runtime/RuntimePrivilegedUnitTests_Run_Tests/

Failure in unit tests:

```
	 panic: assignment to entry in nil map
	 
	 goroutine 59 [running]:
	 github.com/cilium/cilium/pkg/fqdn/dnsproxy.(*DNSProxy).ServeDNS(0xc0005fcc80, 0x20d82e0, 0xc0000b3260, 0xc00011e240)
	 	/home/vagrant/go/src/github.com/cilium/cilium/pkg/fqdn/dnsproxy/proxy.go:642 +0x2418
	 github.com/miekg/dns.(*Server).serveDNS(0xc000484690, 0xc0004f4f00, 0x1b, 0x1b, 0xc0000b3260)
	 	/home/vagrant/go/src/github.com/cilium/cilium/vendor/github.com/miekg/dns/server.go:597 +0x20b
	 github.com/miekg/dns.(*Server).serveTCPConn(0xc000484690, 0xc000512dc0, 0x20d8700, 0xc000010bc8)
	 	/home/vagrant/go/src/github.com/cilium/cilium/vendor/github.com/miekg/dns/server.go:520 +0x205
	 created by github.com/miekg/dns.(*Server).serveTCP
	 	/home/vagrant/go/src/github.com/cilium/cilium/vendor/github.com/miekg/dns/server.go:432 +0x329
	 FAIL	github.com/cilium/cilium/pkg/fqdn/dnsproxy	0.050s
	 FAIL
	 Makefile:97: recipe for target 'tests-privileged' failed
	 make: Leaving directory '/home/vagrant/go/src/github.com/cilium/cilium'
```

--- a/pkg/fqdn/dnsproxy/proxy_test.go
+++ b/pkg/fqdn/dnsproxy/proxy_test.go
@@ -650,6 +650,7 @@ func (s *DNSProxyTestSuite) TestFullPathDependence(c *C) {
 	c.Assert(restored3, checker.DeepEquals, expected3)
 
 	// Test with limited set of allowed IPs
+	oldUsed := s.proxy.usedServers
 	s.proxy.usedServers = map[string]struct{}{"127.0.0.2": {}}
 
 	expected1b := restore.DNSRules{
@@ -668,7 +669,7 @@ func (s *DNSProxyTestSuite) TestFullPathDependence(c *C) {
 	c.Assert(restored1b, checker.DeepEquals, expected1b)
 
 	// unlimited again
-	s.proxy.usedServers = nil
+	s.proxy.usedServers = oldUsed
 
 	s.proxy.UpdateAllowed(epID1, 53, nil)
 	s.proxy.UpdateAllowed(epID1, 54, nil)

